### ISSUE TITLE
Events: display address in "Where" field for all Events at VA facilities instead of linking to facility's web page

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#13222]
### User story

**AS A** Veteran
**I WANT** all events at VA facilities to list the address of the event on the event page
**SO THAT** I have a consistent experience and don't have to go to another page to see the address.

Implements [Sketch file](https://www.sketch.com/s/8ec18b80-a905-47fb-a217-5720450befa1) of design from #13124. (Sketch file should open in the "Location field revisions" tab.)

### Design details

All events will show a similarly formatted address field for the "Where" field. 
- All events will show the street address
- All events will offer a "Directions (Google Maps)" link below the address
- All events will show the building/floor/room field above the address if present (it's optional in the CMS)

VA-facility events additionally will show 
- The name of the VA facility above the info mentioned above
- The name will be hyperlinked to the facility's dedicated web page _if it exists_ – note: some will not exist, e.g. Veteran Outstations

The Sketch file shows versions with and without the optional b/f/r field.

#### VA Facility event

<details>
<summary>VA Facility mockups</summary>

Note the presence of the VA facility name on first line of Where field:

<img width="495" alt="image" src="https://user-images.githubusercontent.com/4054752/236063959-913fc942-22ee-4663-b527-81d106d35a29.png">

</details>

#### Non-VA Facility event

Note that (optional) building/floor/room is first line of Where field:

<details>
<summary>Non-VA Facility mockups</summary>
<img width="521" alt="image" src="https://user-images.githubusercontent.com/4054752/236063335-72e682fb-794e-4e44-b977-62c78cb8d303.png">

</details>

### Engineering notes / background

How to avoid linking VA Facility types that don't have web pages:
- hard-code a list of the 2-3 facility content types that don't have dedicated web pages
- may have to add content type to GQL query

Google directions link
- talk to Ryan or Swirt about what data Facility Locator passes to Google

### Quality / testing notes
Need to verify: 
* Address appears in all cases for non-virtual events (whether VA facility or not)
* Facility link appears for facilities with a web page
* Facility link does _not_ appear for facilities that do not have a web page

- [ ] **Non-VA facility event** location displays address / map link
- [ ] **VA Facility event locations, if Facility has a web page:** Facility name links to Facility page, address appears, Google Map link appears / works
    - [ ] VAMC single event
    - [ ] VAMC events listing
    - [ ] Vet Center single event
    - [ ] Vet Center events listing -- this listing page will include Vet Center outstations, Vet Center CAPs that may be facilities with no Facility link
    - [ ] Outreach events single event - some Outreach Events may be virtual, or may be at facilities with no web page
    - [ ] Outreach events listing - some Outreach Events may be virtual, or may be at facilities with no web page
- [ ] **VA Facility event locations, if Facility _does not_ have a web page**: Facility name _**does not**_ link to Facility page, address appears, Google Map link appears / works
    - [ ] Vet Center outstation single event
    - [ ] Vet Center CAP single event
    - [ ] Mobile Vet Center single event
    - [ ] NCA Facility single event (Facilities exist in CMS, do not have street addresses in CMS, but Facilities is adding address data in Sprint 84; check with Swirt re: status. This may or may not be done enough to test for our use case by the time this ticket is assigned.)
    - [ ] VBA Facility single event (Facilities exist in CMS, have addresses, but do not currently have web pages. They will in the future.)
- [ ] Virtual location events: do not display address data

## Acceptance criteria

- [x] Address is presented as described above, including Directions link 
- [x] Optional building/floor/room text is displayed if provided by CMS
- [x] Google maps directions link is present / working for all addresses
- [x] VA facilities additionally display the name of the VA facility
  - [x] If the VA facility has a web page (eg VAMC), the facility name is hyperlinked to that page (see Eng Notes above)
  - [x] Otherwise, there is no hyperlink (eg VC Outstation)
- [x] Design review required
- [x] A11y review required
- [x] Tests are written and passing for coverage as determined by team/reviewers

**Comment by swirtSJW** [comment]
Concern at the moment is that if someone picks an NCS facility, we don't have the address in the CMS yet.  (coming in a sprint or 2, but not yet.

**Comment by swirtSJW** [comment]
> talk to Ryan or Swirt about what data Facility Locator passes to Google

These are just query parameters in the link that relate to the address.  No big deal.

**Comment by swirtSJW** [comment]
One concern for me that bubbled up when this came up in Facilities Scrum is around the idea of being able to select a Mobile Vet Center.  The address we have for a mobile vet center is more likely where they park it at night, than where it is on any given day.  So either of two optoons.
- Simplest  - Remove the option (if it exists) to select a Mobile Vet Center (they are a different content type)   That way they have to put in a custom address for where the VetCenter will be having this event.
- More difficult with no gain for either editor or Veteran,   Let them pick a mobile vet center, but then force and address field to appear that is required.

**Comment by jilladams** [comment]
> Concern at the moment is that if someone picks an NCA facility, we don't have the address in the CMS yet. (coming in a sprint or 2, but not yet.
This was my mistake. Thanks for clarifying!

**Comment by jilladams** [comment]
Decision from Wes / Jill / DaveC discussion, from DaveC: 
* Treat all VA Facilities the same for this iteration, knowing that mobile locations may be strange. 
* After ship, we can re-evaluate and specifically address the needs of Mobile vet centers, talk to some owners, and plan for a future iteration.
* Will become a Facilities problem for how to deal with Vet Center events, in general, along with Domiciliaries, Community living centers, mobile clinics. (FYI @kmariepat-cityfriends )

**Comment by stefaniefgray** [comment]
Note: If a Vet Center Outstation is listed as an event location as being a "VA Facility," this creates a broken link (see original tickets on this subject #10465 and #13124) due to the fact that Outstations do not have their own URLs

This created a broken link on the following page: https://prod.cms.va.gov/node/58518

Same issue is also happening when linking to published Vet Center nodes -- which _do_ have their own URLs

Broken link was created on https://prod.cms.va.gov/outreach-and-events/events/58456 by linking to the [Little Rock Vet Center - North Little Rock](https://prod.cms.va.gov/little-rock-vet-center/community-access-point/little-rock-vet-center-north-little-rock)

Would recommend bumping up in priority due to large number of Outreach Hub events being created at Vet Center facilities

**Comment by swirtSJW** [comment]
This still needs an AC adjustment for Mobile facilities because the address for a mobile facility is the garage where they park it at night, not where they are
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/5752113/a8a1d726-b0b5-4362-8d85-8d02d6dd420b)

**Comment by dsasser** [comment]
Discussions around how to handle mobile centers were discussed further in [slack](https://dsva.slack.com/archives/C52CL1PKQ/p1692721491718319).

tl;dr: it is relatively trivial to filter out mobile vet centers and clinics such that they cannot be selected using the "At a VA Facility" location. This would require that mobile centers users chose 'At a non-VA location', which requires an address be entered.

I propose we:

-  Add a new CMS BE ticket to update the Entity Reference View "Entity Reference: Systems" to filter out mobile vet centers and clinics. See [this](https://dsva.slack.com/archives/C52CL1PKQ/p1692721891820019?thread_ts=1692721491.718319&cid=C52CL1PKQ) for details from Swirt on how to do that.
-  Decide if the BE ticket should block this ticket.

**Comment by chriskim2311** [comment]
@stefaniefgray @swirtSJW Working through this ticket and was wondering if we still need to account for the potentially broken links for Outstations? If so do we have an idea of which facility content types we want to exclude from providing the clickable URL.

**Comment by chriskim2311** [comment]
@thejordanwood For this ticket we are implementing a link to google maps. We have this text for the link already in the drupal content-build code base **Get directions on Google Maps**.  Should we follow this or use the new text in the designs  **Directions(Google Maps)**

**Comment by thejordanwood** [comment]
@chriskim2311 Good catch! The link should be **Get directions on Google Maps**. The [Sketch file](https://www.sketch.com/s/8ec18b80-a905-47fb-a217-5720450befa1) (Location field revisions tab) should show it this way.

For context: Around the time this design was made, we decided to start changing text from **Directions(Google Maps)** to **Get directions on Google Maps** to make the language clearer. This aligns with the [newer guidelines](https://design.va.gov/content-style-guide/links#linking-to-external-sites) for linking to external sites.

**Comment by chriskim2311** [comment]
Code has been merged for content-build and vets-website. Will verify once its live on production.

**Comment by chriskim2311** [comment]
@jilladams @wesrowe Upon checking production I saw a slight issue with some CSS styles when looking at some of the event detail pages. I have a [PR](https://github.com/department-of-veterans-affairs/content-build/pull/1691) out for the fix

**Comment by chriskim2311** [comment]
Verified that events are showing all ACs and now shows correct formatting for all types of events. Closing!
